### PR TITLE
[onert/nnpackage_run] Add an option `--parallel_inputs`

### PR DESCRIPTION
**Pull request by ragmani**
This commit adds an option `--parallel_inputs`.

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

Related issue: #9207

### CODE REVIEW

File: tests/tools/nnpackage_run/src/nnpackage_run.cc [hunk 1]
```diff
@@ -206,6 +224,22 @@ int main(const int argc, char **argv)
       RandomGenerator(session).generate(inputs);
 #endif
 
+    const auto &parallel_inputs = args.getParallelInputs();
+    const auto backend_list = getAvailableParallelBackendList(available_backends);
+    if (parallel_inputs.size() != 0)
+    {
+      // Check whether supported backends is used
+      if (backend_list.size() != 1 || (backend_list.size() != 0 && backend_list.at(0) != "trix"))
```
Comment (ragmani) [inline tests/tools/nnpackage_run/src/nnpackage_run.cc:232]: I'm not sure if `nnpackage_run` has a dependency on the specific backend name `trix`.

Comment (ragmani) [reply tests/tools/nnpackage_run/src/nnpackage_run.cc:232]: OK, I will move this exception handling into `onert`'s loaders.

File: runtime/onert/backend/trix/DevContext.cc [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  auto device_count = getnumNPUdeviceByType(NPUCOND_TRIV2_CONN_SOCIP);
+  if (device_count <= 0)
+  {
+    throw std::runtime_error("Unable to find TRIX NPU device");
+  }
+
+  int32_t device_num = batch_num == ContextData::no_batch_parallel ? 0 : batch_num % device_count;
+
+  if (getNPUdeviceByType(&_dev_handle, NPUCOND_TRIV2_CONN_SOCIP, device_num) < 0)
```
Comment (chunseoklee) [inline runtime/onert/backend/trix/DevContext.cc:44]: This seems to get only one npudev handle.

Comment (chunseoklee) [reply runtime/onert/backend/trix/DevContext.cc:44]: > This seems to get only one npudev handle. 

I forgot that execution is controlled by execution class.

**Comment by chunseoklee** [comment]
@ragmani I got a result on real device based on commit ed0a88f0609d4a91187e5025fb55d75cdf650f5c

```
tv@tv ~/chunseoklee> sdb2 shell BACKENDS=trix /usr/bin/nnfw-test/Product/out/bin/nnpackage_run
  --parallel_inputs="[0,[true,false,false,false]]"  --shape_run="[0,[10,224,224,3]]" /opt/trix_
batch/mv1.q8/ --output_sizes="[0,10010]"
Package Filename /opt/trix_batch/mv1.q8/
Compileted executor's batch number : 0
Compileted executor's batch number : 1
Compileted executor's batch number : 2
Compileted executor's batch number : 3
Compileted executor's batch number : 4
Compileted executor's batch number : 5
Compileted executor's batch number : 6
Compileted executor's batch number : 7
Compileted executor's batch number : 8
Compileted executor's batch number : 9
===================================
MODEL_LOAD   takes 2.107 ms
PREPARE      takes 16.066 ms
EXECUTE      takes 405.453 ms
- MEAN     :  405.453 ms
- MAX      :  405.453 ms
- MIN      :  405.453 ms
- GEOMEAN  :  405.453 ms
===================================
```
And there seems a problem when running multiple times :

```
v@tv ~/chunseoklee> sdb2 shell BACKENDS=trix /usr/bin/nnfw-test/Product/out/bin/nnpackage_run   --parallel_inputs="[0,[true,false,false,false]]"  --shape_run="[0,[10,224,224,3]]" /opt/trix_batch/mv1.q8/ --output_sizes="[0,10010]" -w 1 -r 2
Package Filename /opt/trix_batch/mv1.q8/
Compileted executor's batch number : 0
Compileted executor's batch number : 1
Compileted executor's batch number : 2
Compileted executor's batch number : 3
Compileted executor's batch number : 4
Compileted executor's batch number : 5
Compileted executor's batch number : 6
Compileted executor's batch number : 7
Compileted executor's batch number : 8
Compileted executor's batch number : 9
Error during nnfw_session::run : std::future_error: No associated state

```

**Comment by ragmani** [comment]
> @ragmani I got a result on real device based on commit https://github.com/Samsung/ONE/commit/ed0a88f0609d4a91187e5025fb55d75cdf650f5c

It was fixed on commit https://github.com/Samsung/ONE/commit/dc06b3f764693962d0862e672a8c9cc6c949883f

**Comment by ragmani** [comment]
@nnfw-bot test x64-builder

**Comment by hseok-oh** [comment]
@nnfw-bot test x64-builder

**Comment by ragmani** [comment]
This draft was discarded.

**Pull request closed without merging**